### ISSUE TITLE
Allow respond_with methods to accept a block

### ISSUE THREAD

**Issue by westonganger** [amberframework/amber#1053]
Currently the `respond_with` format methods only allow a String or Proc. The Proc syntax is awkward and unexpected. It is also currently undocumented.

```ruby
respond_with do
  html -> {
    flash[:notice] = "Sucesss"
    render("my_action.slang")
  }
end
```

It would be alot nicer if we could do simply pass a Block like the following:

```ruby
respond_with do
  html do
    flash[:notice] = "Sucesss"
    render("my_action.slang")
  end
end
```

I tried creating a patch:

```
module Amber::Controller::Helpers::Responders
  def html(&block)
    @available_responses[TYPE[:html]] = block.call; self
  end
end
```

However I kept running into the following error:

```
undefined local variable or method 'html'

      html do
      ^~~~
```

**Comment by elorest** [comment]
I'll look into this.

**Comment by elorest** [comment]
I've submitted a PR to address this.

**Comment by robacarp** [comment]
@elorest @westonganger fixed?

**Comment by Sija** [comment]
@robacarp yep, fixed by #1054.
